### PR TITLE
Redux cart

### PR DESCRIPTION
**Pull request by vrx29**
This PR contains code for Cart and Products data migration from Context API to Redux toolkit.

### CODE REVIEW

File: src/components/ProductCard/index.js [hunk 1]
```diff
@@ -31,30 +33,30 @@ export function ProductCard({ product }) {
           </div>
         </div>
         <div className="card-footer">
-          {cart.some((item) => item._id === product._id) ? (
+          {cart?.some((item) => item._id === product._id) ? (
```
Comment (mikesch803) [inline src/components/ProductCard/index.js:36]: suggestion : you can make this function and move this to utils folder

**Review by mikesch803 — APPROVED** [review]
Nothing more to suggest, you have written good code!

**Review by b0llu — APPROVED** [review]
Great work on redux. Nothing to suggest you can merge this PR.

File: src/redux/services/productServices.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export const getProductsService = async () => {
+  return await axios.get("/api/products");
```
Comment (keyurshah339) [inline src/redux/services/productServices.js:4]: add try catch block

**Review by keyurshah339 — COMMENTED** [review]
can make api calling more structured

rest looks clean

File: src/redux/services/productServices.js (hunk 2 above)
Comment (vrx29) [reply src/redux/services/productServices.js:4]: added while calling. Its just a service

**Pull request merged**